internal/pkg/propagation/summaries.go

```diff
@@ -0,0 +1,761 @@
+	ifTainted:   0b1,
+	taintedRets: []int{0},
+}
+
+// funcSummaries contains summaries for regular functions
+// that could be called statically.
+var funcSummaries = map[string]summary{
+	// func Errorf(format string, a ...interface{}) error {
+	"fmt.Errorf": {
+		ifTainted:   0b11,
+		taintedRets: []int{0},
+	},
+	// func Sprint(a ...interface{}) string {
+	"fmt.Sprint": fromFirstArgToFirstRet,
+	// func Sprintf(format string, a ...interface{}) string {
+	"fmt.Sprintf": {
+		ifTainted:   0b11,
+		taintedRets: []int{0},
+	},
+	// func Sprintln(a ...interface{}) string {
+	"fmt.Sprintln": fromFirstArgToFirstRet,
+	// func Fprint(w io.Writer, a ...interface{}) (n int, err error) {
+	"fmt.Fprint": {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// func Fprintf(w io.Writer, format string, a ...interface{}) (n int, err error) {
+	"fmt.Fprintf": {
+		ifTainted:   0b110,
+		taintedArgs: []int{0},
+	},
+	// func Fprintln(w io.Writer, a ...interface{}) (n int, err error) {
+	"fmt.Fprintln": {
+		ifTainted:   0b10,
+		taintedArgs: []int{0},
+	},
+	// func Sscan(str string, a ...interface{}) (n int, err error) {
+	"fmt.Sscan": {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// func Sscanln(str string, a ...interface{}) (n int, err error) {
+	"fmt.Sscanln": {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// func Sscanf(str string, format string, a ...interface{}) (n int, err error) {
+	"fmt.Sscanf": {
+		ifTainted:   0b1,
+		taintedArgs: []int{2},
+	},
+	// func Fscan(r io.Reader, a ...interface{}) (n int, err error) {
+	"fmt.Fscan": {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// func Fscanln(r io.Reader, a ...interface{}) (n int, err error) {
+	"fmt.Fscanln": {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// func Fscanf(r io.Reader, format string, a ...interface{}) (n int, err error) {
+	"fmt.Fscanf": {
+		ifTainted:   0b1,
+		taintedArgs: []int{2},
+	},
+	// func New(text string) error {
+	"errors.New": fromFirstArgToFirstRet,
+	// func Unwrap(err error) error {
+	"errors.Unwrap": fromFirstArgToFirstRet,
+	// func As(err error, target interface{}) bool {
+	"errors.As": {
+		ifTainted:   0b1,
+		taintedArgs: []int{1},
+	},
+	// func SplitN(s, sep string, n int) []string {
+	"strings.SplitN": fromFirstArgToFirstRet,
+	// func SplitAfterN(s, sep string, n int) []string {
+	"strings.SplitAfterN": fromFirstArgToFirstRet,
+	// func Split(s, sep string) []string {
+	"strings.Split": fromFirstArgToFirstRet,
+	// func SplitAfter(s, sep string) []string {
+	"strings.SplitAfter": fromFirstArgToFirstRet,
+	// func Fields(s string) []string {
+	"strings.Fields": fromFirstArgToFirstRet,
+	// func FieldsFunc(s string, f func(rune) bool) []string {
+	"strings.FieldsFunc": fromFirstArgToFirstRet,
+	// func Join(elems []string, sep string) string {
+	"strings.Join": {
+		ifTainted:   0b11,
+		taintedRets: []int{0},
+	},
+	// func Map(mapping func(rune) rune, s string) string {
+	"strings.Map": {
+		ifTainted:   0b10,
```

Comment:
Do we have any concept of a tainted `func`?  I suppose we could contrive one by injecting a source through a closure as a freevar, but that's probably well out of scope for us.  (Also applicable to many `func` below.)

Reply:
This is a good question. I agree that this is out of scope, without pointer analysis I don't think we could model this with good enough precision.